include/exadg/functions_and_boundary_conditions/interface_coupling.h

```diff
@@ -533,175 +142,70 @@ class InterfaceCoupling
-                                                               marked_vertices,
-                                                               cache_src));
-
-      InterfaceCommunicator<dim, dim> & communicator = map_communicator.find(quadrature)->second;
-
-      mpi_map_q_points_src = communicator.get_remote_quadrature_points();
-
-      communicator.init_solution_values(array_solution_dst);
-
-      communicator.init_remote_solution_values(mpi_map_solution_src);
-
-
-      /*
-       * 3. Compute dof indices and shape values for all quadrature points
-       */
-      typename Triangulation<dim, dim>::active_cell_iterator cell_hint =
-        triangulation_src->begin_active();
-
-      for(auto it : mpi_map_q_points_src)
-      {
-        mpi_rank const          proc               = it.first;
-        ArrayQuadraturePoints & array_q_points_src = it.second;
-
-        mpi_map_cache_src.emplace(proc, ArrayVectorCache());
-        ArrayVectorCache & array_cache_src = mpi_map_cache_src.find(proc)->second;
-        array_cache_src.resize(array_q_points_src.size());
-
-        for(types::global_dof_index q = 0; q < array_q_points_src.size(); ++q)
-        {
-          auto adjacent_cells =
-            find_all_active_cells_around_point(*mapping_src,
-                                               dof_handler_src->get_triangulation(),
-                                               array_q_points_src[q],
-                                               tolerance,
-                                               cell_hint,
-                                               marked_vertices,
-                                               *cache_src);
-
-          array_cache_src[q] = get_dof_indices_and_shape_values(adjacent_cells,
-                                                                *dof_handler_src,
-                                                                *mapping_src,
-                                                                *dof_vector_src_double_ptr);
-
-          AssertThrow(array_cache_src[q].size() > 0,
-                      ExcMessage("No adjacent points have been found."));
-        }
-      }
-
-
-      /*
-       * 4. Communication: transfer results back to dst-side
-       */
-      communicator.process(mpi_map_solution_src, array_solution_dst, 11 + quadrature);
+      map_communicator.emplace(
+        quadrature,
+        Utilities::MPI::RemotePointEvaluation<dim>(tolerance, false, 0, [marked_vertices]() {
+          return marked_vertices;
+        }));
+
+      map_communicator[quadrature].reinit(array_q_points_dst,
+                                          dof_handler_src_in.get_triangulation(),
+                                          mapping_src);
     }
 
     // finally, give boundary condition access to the data
     for(auto boundary : map_bc)
     {
       boundary.second->set_data_pointer(map_index_dst, map_solution_dst);
     }
+
+    update_data(dof_vector_src_in); // TODO: needed?
```

Comment:
I think the name `setup` suggests we do not exchange any data, but I think @nfehn should decide.

Reply:
@peterrum Did you introduce this because you removed the communication step above? Or are these two independent steps?

I would argue that setup prepares everything so that the boundary conditions can be used/evaluated. This would imply that a DoF-vector handed over to the function is indeed evaluated. If the "exchange of data" is done in the old code, we should also do it here in my opinion, in order to not introduce bugs. Right now, I do not remember all the details of the implementation.

Reply:
> Did you introduce this because you removed the communication step above? Or are these two independent steps?

Yes this is the replacement of the old communication step. I am still wondering what the purpose of it is. I guess to set the initial values. But why is it done during initialization of the communication class and not in a separate call?

Reply:
I think this originates from different perspectives. From my point of view, `InterfaceCoupling` is more than the communication class. In other words, the communication class is (one, the main, ...) part of the `InterfaceCoupling`. Hence, `InterfaceCoupling::setup()` wants to make sure that everything is initialized. So for now I believe we should keep this definition, but in principle we could also redefine it.